### PR TITLE
Add Ledger WebHID support

### PR DESCRIPTION
**Pull request by millsmcilroy**
Closes https://github.com/shapeshift/hdwallet/issues/385

### CODE REVIEW

File: examples/sandbox/index.html [hunk 1]
```diff
@@ -8,409 +8,410 @@
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css" />
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/milligram/1.3.0/milligram.min.css" />
     <style>
-      .container {
-        margin-top: 2em;
-        padding: 1em;
-        border: #ccc solid 1px;
-        border-radius: 6px;
-      }
+        .container {
```
Comment (millsmcilroy) [inline examples/sandbox/index.html:11]: Need to get prettier to format this file. It's currently ignoring for some reason

File: packages/hdwallet-ledger-webhid/src/adapter.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import * as core from "@shapeshiftoss/hdwallet-core";
+import * as ledger from "@shapeshiftoss/hdwallet-ledger";
+
+import { LedgerWebHIDTransport, getFirstLedgerDevice, getTransport, openTransport } from "./transport";
```
Comment (reidrankin) [inline packages/hdwallet-ledger-webhid/src/adapter.ts:4]:
```suggestion
import { LedgerWebHIDTransport, MOCK_SERIAL_NUMBER, getFirstLedgerDevice, getTransport, openTransport } from "./transport";
```

File: packages/hdwallet-ledger-webhid/src/transport.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+import * as core from "@shapeshiftoss/hdwallet-core";
+import * as ledger from "@shapeshiftoss/hdwallet-ledger";
+import {
+  LedgerResponse,
+  LedgerTransportCoinType,
+  LedgerTransportMethod,
+  LedgerTransportMethodName,
+} from "hdwallet-ledger/src/transport";
+
+const RECORD_CONFORMANCE_MOCKS = false;
```
Comment (reidrankin) [inline packages/hdwallet-ledger-webhid/src/transport.ts:17]:
```suggestion
const RECORD_CONFORMANCE_MOCKS = false;

export const MOCK_SERIAL_NUMBER = "ledger-webhid-device" // WebHID devices do not have serialNumbers
```

File: packages/hdwallet-ledger-webhid/src/adapter.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      }
+    }, APP_NAVIGATION_DELAY);
+  }
+
+  public get(): core.HDWallet {
+    return core.mustBeDefined(this.keyring.get(MOCK_SERIAL_NUMBER));
+  }
+
+  // without unique device identifiers, we should only ever have one HID ledger device on the keyring at a time
+  public async initialize(device?: HIDDevice | null): Promise<number> {
```
Comment (reidrankin) [inline packages/hdwallet-ledger-webhid/src/adapter.ts:69]:
```suggestion
  public async initialize(device?: HIDDevice): Promise<number> {
```

**Review by reidrankin — APPROVED** [review]
here's a bunch of bikeshedding BS, take it or leave it

**Comment by 0xean** [comment]
@millsmcilroy - can you please comment on if you plan to make these changes suggested by @mrnerdhair?

We would like to get this merged as we have a lot of PRs stacking up.   Thank you!

**Pull request merged**